incubator/kafka/templates/StatefulSet.yaml

```diff
@@ -1,55 +1,3 @@
-# headless service for petset DNS
-apiVersion: v1
-kind: Service
-metadata:
-  name: "{{ printf "%s-zookeeper-%s" .Release.Name .Values.zookeeper.Name | trunc 63 }}"
-spec:
-  ports:
-  - port: 2181
-    name: client
-  selector:
-    component: "{{.Release.Name}}"
-
-# A headless service to create DNS records
----
-apiVersion: v1
-kind: Service
-metadata:
-  name: "{{ printf "%s-broker-%s" .Release.Name .Values.Name | trunc 63 }}"
-  labels:
-    heritage: {{.Release.Service | quote }}
-    release: {{.Release.Name | quote }}
-    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
-    component: "{{.Release.Name}}-{{.Values.Component}}"
-  annotations:
-    "helm.sh/created": {{.Release.Time.Seconds | quote }}
-    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
-spec:
-  ports:
-  - port: 9092
-  clusterIP: None
-  selector:
-    component: "{{.Release.Name}}-{{.Values.Component}}"
-    app: kafka
----
-apiVersion: v1
-kind: Service
-metadata:
-  name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 63 }}"
-  labels:
-    heritage: {{.Release.Service | quote }}
-    release: {{.Release.Name | quote }}
-    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
-    component: "{{.Release.Name}}-{{.Values.Component}}"
-  annotations:
-    "helm.sh/created": {{.Release.Time.Seconds | quote }}
-spec:
-  ports:
-  - port: 9092
-  selector:
-    component: "{{.Release.Name}}-{{.Values.Component}}"
-    app: kafka
----
```

Comment:
Additionally, one issue we have seen when we attempted to use this chart as a basis of work was that the Kafka cluster internally caches the ZK IPs due to a prevalent issue with the ZK client SDKs. The impact of this is that if you do a rolling up grade of your ZK cluster Kafka will stop functioning as it is still holding onto all the old IPs.

I don't know that this can be addressed really with this Chart, however it is a limitation of this deployment that people should be aware of. We addressed this by creating N ZK services (essentially a VIP at this point that is static) and then tagging each ZK node in the stateful set with it's own unique label. That way the ZK-1 service IP is consistent but the underlying pod can still be upgraded without breaking all the clients.